build.gradle.kts

```diff
@@ -1,11 +1,13 @@
 buildscript {
     extra.apply {
-        set("version_kotlin", "1.7.10")
-        set("version_android_gradle_plugin", "7.4.2")
+        set("version_kotlin", "1.8.21")
+        set("version_android_gradle_plugin", "8.0.2")
         set("version_google_services", "4.3.14")
-        set("version_compose", "1.3.1")
+        set("version_compose", "1.4.3")
+        set("version_compose_compiler", "1.4.7")
         set("version_compose_nav", "2.4.2")
-        set("version_hilt", "2.45")
+        set("version_compose_activity", "1.7.2")
+        set("version_hilt", "2.46.1")
```

Comment:
What's the purpose of updating all these dependencies in this PR?

Reply:
Mainly to be compatible with the upgrade of Kotlin and the upgrade required for this version of KMM. with the current version gave me compile errors. The bug is fixed between 2.46 and 2.46.1
https://github.com/google/dagger/releases